dialer.go

```diff
@@ -9,19 +9,28 @@ import (
 // to the message bus.
 type Dialer func(string, string) (net.Conn, error)
 
-// DefaultDialer is the RPC server default implementation of
-// a dialer.
-func DefaultDialer(network, addr string) (net.Conn, error) {
-	conn, err := net.DialTimeout(network, addr, 2*time.Second)
-	if err != nil {
-		return nil, err
-	}
-	// Heartbeating hasn't started yet, don't stall forever on a dead server.
-	// A deadline is set for TLS and AMQP handshaking. After AMQP is established,
-	// the deadline is cleared in openComplete.
-	if err := conn.SetDeadline(time.Now().Add(2 * time.Second)); err != nil {
-		return nil, err
-	}
+// DialConf is a structure that you can pass to the default dialer to
+// change the defautl timeouts for dailing and handshakes.
+type DialConf struct {
+	DialTimeout time.Duration
+	Deadline    time.Duration
+}
 
-	return conn, nil
+// DefaultDialer will return the default RPC server default implementation of
+// a dialer. With the DialConf parameter the default timeous can be specified.
+func DefaultDialer(config DialConf) Dialer {
+	return func(network, addr string) (net.Conn, error) {
+		conn, err := net.DialTimeout(network, addr, config.DialTimeout)
+		if err != nil {
+			return nil, err
+		}
+		// Heartbeating hasn't started yet, don't stall forever on a dead server.
+		// A deadline is set for TLS and AMQP handshaking. After AMQP is established,
+		// the deadline is cleared in openComplete.
+		if err := conn.SetDeadline(time.Now().Add(config.Deadline)); err != nil {
```

Comment:
amqp091's default dialer has this hard coded to 2s i think

Reply:
At least it was when we copied this code a couple of years ago :D

Reply:
Actually, looks like the current version is exported and does exactly what we do, maybe we can just use that one now? https://github.com/rabbitmq/amqp091-go/blob/34f6f315b4af699bddc44dbe9a012e6a7c415135/connection.go#L115-L132

Reply:
Alright then! Perfect!

Reply:
Perhaps the we can remove the `DialConf`, having only a `dialTimeout` passed to the `DefaultDialer` function.